### ISSUE TITLE
New User questions...

### ISSUE THREAD

**Issue by KeithProctor** [martinblech/xmltodict#327]
Love that you created this tool so TY up front.  I'm having the following issues:

1. How do you not add the # and @ characters being prepended to the name of a name value pair.
2. Is there a way to efficiently deal with paths?  I want to do something like this.

I know this is invalid... Just showing a concept here:

`basePath	= ['FMSaveAsXML']
structPath	= [basePath]['Structure']
layPath		= structPath['LayoutCatalog']   # want an iterable pointer to this location

print ( 'Ver: ' + str(basePath['@version']))
print ( 'layout: ' + str(layPath[0]['LayoutName']))
`

**Comment by KeithProctor** [comment]
@martinblech I've created a function to walk over a list.  The call simplifies things but I consider the function itself a hack.  It's ~30 lines long.  So I do something like:

`GetXMLData( xml, basePath + ['path2', 'path3'])`

So the real question now is... is there a better way?

**Comment by bfontaine** [comment]
Please share a sample XML so we can reproduce your issue.

**Comment by KeithProctor** [comment]
It's just standard XML.  I'm asking is there a better way to request a path into
the dictionary items.

Say I want this path /path1/path2/path3/someField

Python doesn't allow

	path = /path1/path2/path3/someField
	theDict[path]
or
	path = /path1/path2/path3/
	theDict.getLocation(path)

The desire here is to simply my code and either get the field or XML 
block at those locations.  If I know I'll be inside a block for a while
this can speed up my code and simplify usage as well as make the
code more readable.

In short... Is there another way to make the request.  Does your package offer
up another method?

**Comment by bfontaine** [comment]
(I’m just another user of `xmltodict`, I’m not the maintainer)

If your XML contains only a single field of each tag, you could use `theDict["path1"]["path2"]["path3"]`. If it doesn’t or you don’t know in advance, it becomes tricky because each intermediate element can be either be a dict or a list of dicts.

As far as I know, xmltodict just parses the XML into a dict; it doesn’t offer any way to traverse that dict. There are other libraries that offer that kind of feature ([lxml](https://lxml.de/xpathxslt.html#xpath) is probably the most popular) but if you already use xmltodict it might be overkill to change libraries just for that. If you search for "nested dict" on StackOverflow you might be able to find a suitable solution (see [here](https://stackoverflow.com/q/71460721/735926) for example).

**Comment by KeithProctor** [comment]
kk TY for the response.  I like that page but I'm also looking for a response to that last example.

key = ("a", "b", "c")
The objective now is to create some function that works as follows:

>>> getitem(d, key)
0
This format can also conveniently be applied directly as the __getitem__ method of a class.

One more constraint: I want the function to fail noisily when it is asked to get a non-existing key.

Not throwing an exception but returning standard errors like invalid key.

I'll keep looking TY.
